### PR TITLE
Rxns with layers

### PR DESCRIPTION
**Pull request by jtgasparik**
Added is_at_surface flag to indicate if phase exists in the surface layer of particle

### CODE REVIEW

File: src/aero_rep_data.F90 [hunk 1]
```diff
@@ -489,26 +493,53 @@ end function get_name
...
 
     !> List of phase ids
     integer(kind=i_kind), allocatable :: phase_ids(:)
     !> Aerosol representation data
     class(aero_rep_data_t), intent(in) :: this
     !> Aerosol phase name
     character(len=*), intent(in) :: phase_name
+    !> Indicates if aerosol phase is at the surface of particle
+    logical, intent(in), optional :: is_at_surface
+    logical, allocatable :: aero_is_at_surf(:)
```
Comment (mattldawson) [inline src/aero_rep_data.F90]: Is this needed?

File: src/aero_rep_data.F90 [hunk 2]
```diff
@@ -489,26 +493,53 @@ end function get_name
...
+        do i_phase = 1, size(this%aero_phase)
+          if (this%aero_phase(i_phase)%val%name().eq. phase_name .and. &
+              aero_is_at_surf(i_phase) .eqv. .true.) then
+            phase_ids(i_instance) = i_phase
+            i_instance = i_instance + 1
+          end if
+        end do
       end if
-    end do
+      if (.not. is_at_surface) then
```
Comment (mattldawson) [inline src/aero_rep_data.F90]: This could just be `else`

File: src/aero_rep_data.F90 [hunk 3]
```diff
@@ -489,26 +493,53 @@ end function get_name
...
-    do i_phase = 1, size(this%aero_phase)
-      if (this%aero_phase(i_phase)%val%name().eq.phase_name) then
-        phase_ids(i_instance) = i_phase
-        i_instance = i_instance + 1
+    if (present(is_at_surface)) then
+      if (is_at_surface) then
+        i_instance = 1
+        do i_phase = 1, size(this%aero_phase)
+          if (this%aero_phase(i_phase)%val%name().eq. phase_name .and. &
+              aero_is_at_surf(i_phase) .eqv. .true.) then
```
Comment (mattldawson) [inline src/aero_rep_data.F90]:
```suggestion
          if (this%aero_phase(i_phase)%val%name().eq. phase_name .and. &
              aero_is_at_surf(i_phase)) then
```

File: src/aero_rep_data.F90 [hunk 4]
```diff
@@ -489,26 +493,53 @@ end function get_name
...
+            i_instance = i_instance + 1
+          end if
+        end do
       end if
-    end do
+      if (.not. is_at_surface) then
+        i_instance = 1
+        do i_phase = 1, size(this%aero_phase)
+          if (this%aero_phase(i_phase)%val%name().eq. phase_name .and. &
+              aero_is_at_surf(i_phase) .eqv. .false.) then
```
Comment (mattldawson) [inline src/aero_rep_data.F90]:
```suggestion
          if (this%aero_phase(i_phase)%val%name().eq. phase_name .and. &
              .not. aero_is_at_surf(i_phase)) then
```

File: src/aero_reps/aero_rep_single_particle.c [hunk 5]
```diff
@@ -437,8 +437,9 @@ void aero_rep_single_particle_print(int *aero_rep_int_data,
   printf("\nAerosol representation id: %d", AERO_REP_ID_);
   printf("\nMax computational particles: %d", MAX_PARTICLES_);
   printf("\nParticle state size: %d", PARTICLE_STATE_SIZE_);
-  printf("\n\n   - Phases -");
   for(int i_layer = 0; i_layer < NUM_LAYERS_; ++i_layer){
+    printf("\nLayer: %d", i_layer);
+    printf("\n\n   - Phases -");
```
Comment (mattldawson) [inline src/aero_reps/aero_rep_single_particle.c:442]: please remove debug output

File: src/rxns/rxn_SIMPOL_phase_transfer.F90 [hunk 6]
```diff
@@ -374,6 +373,7 @@ subroutine initialize(this, chem_spec_data, aero_rep, n_cells)
 
     ! Save the index of the gas-phase species in the state variable array
     GAS_SPEC_ = chem_spec_data%gas_state_id(gas_spec_name)
+    print *, GAS_SPEC_
```
Comment (mattldawson) [inline src/rxns/rxn_SIMPOL_phase_transfer.F90]: please remove debug code

**Review by mattldawson — CHANGES_REQUESTED** [review]
looks great! just a few suggestions

File: src/aero_reps/aero_rep_single_particle.c (hunk 5 above)
Comment (jtgasparik) [reply src/aero_reps/aero_rep_single_particle.c:442]: We added this to the print function to print layer properties. This should have been part of the last PR.

**Review by jcurtis2 — APPROVED** [review]
While it makes it a longer variable name, I think you should at least consider changing `aero_is_at_surface` to `aero_phase_is_at_surface` to more clearly denote what this is and be more consistent with the related variable `aero_phase`.

**Review by mattldawson — APPROVED** [review]
looks great!

**Pull request merged**